src/boot/efi/SECURITY.md

```diff
@@ -82,6 +82,21 @@ and manual confirmation will be asked before proceeding. When running as a virtu
 enrollment is fully automated, without user interaction, unless disabled via a configuration file in the
 ESP. The configuration file can also be used to disable enrollment completely.
 
+## Compiler Hardening
+The PE binaries are built with `-fstack-protector-strong`, and the stack canary is seeded with random data if
+the UEFI RNG protocol is available.
+
+The binaries also are linked with `-z relro` and ship with native PE relocations, with the conversion from
+ELF performed at build time, instead of containing ELF dynamic relocations, so the firmware/Shim image
+loader requires fewer writable pages.
```

Comment:
It's not the loader but the *loaded* image that requires fewer pages marked as writable.